src/UI/Component/DropDown/DropDownItem.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/* Copyright (c) 2017 Alexander Killing <killing@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\DropDown;
+
+use ILIAS\UI\Component\Clickable;
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\Hoverable;
+use ILIAS\UI\Component\JavaScriptBindable;
+
+/**
+ * Drop down items
+ */
+interface DropDownItem extends Component, JavaScriptBindable, Clickable, Hoverable {
```

Comment:
I got your question wrong sorry. On the component you would to a type check, but not on the renderer.

Concerning the context stack. That might be an issue.